### DISCUSSION TITLE
Rich data types in Blocks

### DISCUSSION THREAD

**Discussion by zachlloyd** [warpdotdev/Warp#431] [Product ideas]
The proposed feature would allow for special rendering of structured block types (e.g. json, csv, markdown).

Exact implementation tbd. but I'm interested in whether something like this is appealing to the community.

Here's a sense of what it might look like:

![Warp product concepts enterprise - Google Slides 2021-07-20 at 11 53 48 AM](https://user-images.githubusercontent.com/8727141/126356114-972200ad-b8d4-49db-ba39-3da21897f35a.jpg)

Please upvote if you'd like this.

**Comment by elviskahoro** [comment]
From: [japsu](https://github.com/japsu) - Issue #214 
Please add support for co-operating command line programs to show output as a two-way scrollable table.

Use cases that come to mind that would benefit from this:

Relational database clients such as psql (able to configure output separators in .psqlrc)
numpy, pandas and other Python libraries working with 2D arrays when used with the REPL (conforming output could probably be implemented via a Python module, need not be in numpy/pandas core; or leverage existing support designed for Jupyter Notebook etc. GUI tools)

**Comment by elviskahoro** [comment]
From: hunkybill via Discord: https://discord.com/channels/851854972600451112/954128565178339338/954140096096006224
> Something amazing I would like is along the lines of me tailing some logs. OK. I do that all the time. But my efforts to find stuff is really limited to running a grep... so something that would be nice in Warp would be a way to have a block be "live, have focus, and be something I could setup a filter on"... so that what I am looking for comes through highlighted, but so does the rest of the data. 
> 
> When I do this with a grep, basically I ONLY ever get what passes the grep filter.

**Comment by neongreen** [comment]
https://github.com/nushell/nushell could perhaps serve as inspiration

**Reply by elviskahoro** [reply]
Andres from nushell actually helped Warp get off the ground in the early days!

**Comment by exrhizo** [comment]
Hi, I have a similar use case

My docker logs look like this:
<img width="751" alt="image" src="https://user-images.githubusercontent.com/4206097/194780635-37d13e2d-0114-4049-b7cf-cde744a3285c.png">

But then, not all the json lines have the same fields, so maybe I would need to pipe through a preprocessor to use this feature

Thank you for the great product!

**Reply by exrhizo** [reply]
oh, and I am often streaming the logs and often only care about the last X

**Comment by XiaofengXie16** [comment]
Any updates on this issue? I will even pay for this feature if it is added to warp.

**Reply by i-am-the-slime** [reply]
How much?

**Comment by phgoh** [comment]
might be worth re-exploring again, now that the codebase is open-source
